### DISCUSSION TITLE
download 32 bit images

### DISCUSSION THREAD

**Discussion by marioepsley** [cables-gl/cables_docs#598] [Feature Requests]
wondering, is it or could it be possible to download a 32bit transparent image with cables.

**Comment by pandrr** [comment]
browsers dont support this natively, we could maybe use some library and export exr files from cables. e.g. https://threejs.org/docs/#examples/en/exporters/EXRExporter

**Comment by pandrr** [comment]
how do you want to use those images ?

**Reply by marioepsley** [reply]
only for downloading, so a client doesn't get banding when using blurs etc

**Reply by marioepsley** [reply]
ideally a png

**Comment by wirmachenbunt** [comment]
yeah, i came accross this also a couple of times. is there any way to save/download the texture (format RGBA 32bit float) coming out of here 
![image](https://github.com/user-attachments/assets/4e9c77c9-74f4-4943-a6c5-8b691e9e9ac6)
exr, ktx, dds...anything would be fine. probably exr would be best since cables can load this

**Reply by wirmachenbunt** [reply]
https://threejs.org/docs/#examples/en/exporters/EXRExporter
three.js has this but its way over my head to port this to cables

**Reply by wirmachenbunt** [reply]
thinking about this, i can fit one dev day into the budget for this task. would this work, or is this more complex ?  exr export, 32 bit, to save position texture in order to load them later back into cables

**Comment by pandrr** [comment]
you could use RGBE format, which is not 32bit, but smaller than exr and cables has ops to load and save. 
but yes, would love to create an .exr write op. could be done in a day

**Reply by wirmachenbunt** [reply]
yes, i'm going this way right now, but its a bit tricky because one has to stay within the small area of RGBA 8 bit and negative numbers are not allowed.
https://cables.gl/edit/qPAoDA

one side note, this texture viz doesnt show the texture on my mac for some reason, while the windows machine does.
<img width="1240" alt="image" src="https://github.com/user-attachments/assets/34657557-adab-49da-bf2e-7969704fa39c" />

regarding the developement of EXR export, go for it if you have the time.

**Reply by wirmachenbunt** [reply]
it repaired itself somehow :)
<img width="406" alt="image" src="https://github.com/user-attachments/assets/66ed5641-77e8-4f44-8b47-9b99fd9f6a13" />

**Reply by wirmachenbunt** [reply]
if exr export works for position textures, this would come in really handy. so far i was forced to used houdini or some blender python script. but cables is much more playful to prepare these textures. probably more people will benifit of this, if this feature is known. cables as a EXR authoring tool.
